### DISCUSSION TITLE
How to set or change default shell?

### DISCUSSION THREAD

**Discussion by m8yng** [Azure/CloudShell#335] [Q&A] [answered]
With Version: 12.16.0.1 (dev#3b84051564.230626-2207). The default command-line session is now PowerShell. Is there a recommended way to change this setting rather than switching each time from the upper-left corner dropdown menu?

I was seeking to set this using PowerShell $Profile, but it seems it is not allowed to make changes now.

Any suggestions?

**Comment by theJasonHelmick** [comment, marked as answer]
Hi @m8yng - Thank you for the discussion. Cloud Shell does not support a programmatic way to change from one shell to another.  The left drop down is the method for choosing.  Personally, I normally start a bash session, then when I want PowerShell, I type `pwsh` -- when done, I type `exit` to return to bash.  I understand this might not be what your looking for, but wanted to offer some additional help.

**Reply by m8yng** [reply]
Hi @theJasonHelmick, thanks for replying.
I am using the same trick to switch from `pwsh` to `bash`, and it works nicely.
Just would be nice if the web interface remembers what shell is being used previously.

**Comment by gaborl85** [comment]
To add to this topic, I discovered that you can set the ‘default’ shell when configuring Azure Cloud Shell. Afterward, if you wish to reset the ‘default’ settings you’ve chosen, follow these steps: go to settings (click the wheel icon on the menu bar) and select ‘Reset user settings.’ You’ll be prompted to choose a new default shell.

This information is also mentioned in the documentation, although I believe some additional context would be good. [You can find it here: Azure Cloud Shell Features.](https://learn.microsoft.com/en-us/azure/cloud-shell/faq-troubleshooting).

> Azure Cloud Shell runs on Azure Linux, Microsoft's Linux distribution for cloud infrastructure edge products and services. You can choose Bash or PowerShell as your default shell.

**Reply by m8yng** [reply]
Thanks for the comment.
I also recently discovered that the default shell is set to the last one used in the new user interface.
![](https://learn.microsoft.com/en-us/azure/cloud-shell/get-started/media/ephemeral/choose-shell.png)
doc: [Switch between Bash and Powershell (New UI)](https://learn.microsoft.com/en-us/azure/cloud-shell/new-ui-shell-window#switch-between-bash-and-powershell)
